### PR TITLE
add maxrequestworkers parameter for mpm_worker module

### PR DESCRIPTION
**Pull request by trefzer**

### CODE REVIEW

**Comment by trefzer** [comment]
check seems to fail provisioning SLES-15 which ist not a problem of this commit.

**Review by ekohl — CHANGES_REQUESTED** [review]
In other places (prefork/event mpm) we prefer `MaxRequestWorkers` and don't output `MaxClients` if it's set. Could you implement the same here?

**Comment by trefzer** [comment]
check seems to fail provisioning RedHat 7 and 9, not the change request.

**Comment by david22swan** [comment]
@ekohl Are you happy with the changes made?

File: spec/classes/mod/worker_spec.rb [hunk 1]
```diff
@@ -126,7 +126,8 @@
       it { is_expected.to contain_file('/etc/httpd/conf.modules.d/worker.conf').with(content: %r{^<IfModule mpm_worker_module>$}) }
       it { is_expected.to contain_file('/etc/httpd/conf.modules.d/worker.conf').with(content: %r{^\s+ServerLimit\s+10$}) }
       it { is_expected.to contain_file('/etc/httpd/conf.modules.d/worker.conf').with(content: %r{^\s+StartServers\s+11$}) }
-      it { is_expected.to contain_file('/etc/httpd/conf.modules.d/worker.conf').with(content: %r{^\s+MaxClients\s+12$}) }
+      it { is_expected.not_to contain_file('/etc/httpd/conf.modules.d/worker.conf').with(content: %r{^\s+MaxClients}) }
```
Comment (ekohl) [inline spec/classes/mod/worker_spec.rb:129]: It's better to expect a file to be there without certain content. The big reason is that your example will also pass if the file isn't present in the catalog at all.
```suggestion
      it { is_expected.to contain_file('/etc/httpd/conf.modules.d/worker.conf').without(content: %r{MaxClients}) }
```

On a side note: this whole file should be rewritten IMHO to have a single `it` statement with one long `content` block that has the full generated template, but that's not part of this PR.

File: manifests/mod/worker.pp [hunk 2]
```diff
@@ -8,6 +8,7 @@
 #   The max number of simultaneous requests that will be served.
 #   This is the old name and is still supported. The new name is
 #   MaxRequestWorkers as of 2.3.13.
+#   If maxreuestworkers is set, this value is ignored.
```
Comment (ekohl) [inline manifests/mod/worker.pp:11]:
```suggestion
#   If maxrequestworkers is set, this value is ignored.
```

File: spec/classes/mod/worker_spec.rb (hunk 1 above)
Comment (trefzer) [reply spec/classes/mod/worker_spec.rb]: Ack done

**Review by david22swan — APPROVED** [review]
Little concerned with maxclients no longer having a test where it is explicitly set, but given it has a default value covered in the other tests this should be fine.

LGTM

@ekohl Are you happy that the changes that you requested have been implemented?

File: manifests/mod/worker.pp [hunk 3]
```diff
@@ -40,6 +41,10 @@
 # @param apache_version
 #   Used to verify that the Apache version you have requested is compatible with the module.
 #
+# @param maxrequestworkers
+#   Maximum number of connections that will be processed simultaneously
```
Comment (ekohl) [inline manifests/mod/worker.pp:45]: I think this will show up slightly better in the rendered strings documentation
```suggestion
#   Maximum number of connections that will be processed simultaneously.
```

**Review by ekohl — APPROVED** [review]
One tiny comment, but generally :+1:

**Comment by david22swan** [comment]
Gonna go ahead and merge.
@trefzer Thanks for putting in the work.
@ekohl Thanks for helping to review
